app/js/lib/foxbox/network.js

```diff
@@ -183,19 +186,26 @@ export default class Network extends EventDispatcher {
    * went offline following by another event indicating that box is online
    * again.
    *
+   * @return {Promise}
    * @private
    */
   [p.pingBox]() {
     const previousState = this[p.local] || this[p.remote];
 
-    this[p.ping](`${this.localOrigin}/ping`)
-      .then((isOnline) => this[p.onPong](previousState, p.local, isOnline));
+    const pingPromises = [
+      this[p.ping](`${this.localOrigin}/ping`)
+        .then((isOnline) => this[p.onPong](previousState, p.local, isOnline)),
+    ];
 
     // @todo Find a better way to detect if a tunnel connection is active.
     if (this[p.settings].tunnelOrigin) {
-      this[p.ping](`${this.tunnelOrigin}/ping`)
-        .then((isOnline) => this[p.onPong](previousState, p.remote, isOnline));
+      pingPromises.push(
+        this[p.ping](`${this.tunnelOrigin}/ping`)
+          .then((isOnline) => this[p.onPong](previousState, p.remote, isOnline))
+      );
     }
+
+    return Promise.all(pingPromises);
```

Comment:
> I think we have a similar issue here than when trying to put these 2 pings in a promise.
If one of the ping takes very long to resolve, the next ping will be delayed for as long. 

It's hard to say, preliminary ```ping takes very long to resolve``` should be an edge case, and more important to have less network requests as possible.

> Maybe it's better to have 2 timers here: 1 for local and 1 for remote.
What do you think?

Yeah, maybe we can try that, I'm wondering maybe they should even have different periods... Do you want me to try to do it in this PR or file appropriate issue?